dm/syncer/syncer.go

```diff
@@ -1387,9 +1388,29 @@ func (s *Syncer) syncDDL(queueBucket string, db *dbconn.DBConn, ddlJobChan chan
 
 		if !ignore {
 			var affected int
+			row, err2 := db.QuerySQL(s.syncCtx, s.metricsProxies, "SELECT UNIX_TIMESTAMP()")
+			if err2 != nil {
+				s.tctx.L().Warn("selecting unix timestamp failed", zap.String("error", err2.Error()))
+			}
+			var ddlCreateTime int64
+			for row.Next() {
+				err2 = row.Scan(&ddlCreateTime)
+				if err2 != nil {
+					s.tctx.L().Warn("getting ddlCreateTime failed", zap.String("error", err2.Error()))
+				}
+			}
+			row.Close()
 			affected, err = db.ExecuteSQLWithIgnore(s.syncCtx, s.metricsProxies, errorutil.IsIgnorableMySQLDDLError, ddlJob.ddls)
+			failpoint.Inject("TestHandleSpecialDDLError", func() {
+				err = mysql2.ErrInvalidConn
+				affected = 1
```

Comment:
I have changed it to, when the original index of `ddlJob.ddls` is larger than 2 (the status is injected like "running" instead of caused by real error like "cancelled"), assign `affected` to index minus 2 due to adding the SET SQL of timezone and timestamp.